cmd/main.go

```diff
@@ -73,25 +74,40 @@ var Options struct {
 	DeployTarget                string `envconfig:"DEPLOY_TARGET" default:"k8s"`
 	OCMConfig                   ocm.Config
 	HostConfig                  host.Config
-	LogLevel                    string `envconfig:"LOG_LEVEL" default:"info"`
+	LogConfig                   logconfig.Config
 	LeaderConfig                leader.Config
 }
 
-func main() {
+func InitLogs() *logrus.Entry {
 	log := logrus.New()
 	log.SetReportCaller(true)
 
-	err := envconfig.Process("myapp", &Options)
-	if err != nil {
-		log.Fatal(err.Error())
+	logger := log.WithFields(logrus.Fields{})
+
+	//set log format according to configuration
+	logger.Info("Setting log format: ", Options.LogConfig.LogFormat)
+	if Options.LogConfig.LogFormat == logconfig.LogFormatJson {
```

Comment:
what about setting the format "text" and what if it's an empty string or invalid?